From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: samuelmaddock <samuel.maddock@gmail.com>
Date: Tue, 6 Oct 2020 19:40:32 -0400
Subject: win: disable viz frame throttling

In the refactored Around desktop app, we're rendering content into
remote windows from a hidden background window. This leads to frames
being dropped by the background window.

Disabling the throttling for the case of undrawn frames resolves the
issue and restores rendering to 60fps when using
requestAnimationFrame.

Signed-off-by: gellert <gellert.hegyi@around.co>

diff --git a/components/viz/service/frame_sinks/compositor_frame_sink_support.cc b/components/viz/service/frame_sinks/compositor_frame_sink_support.cc
index 29482a0fc0976bb19544303b7e03bf44d60f07e9..8cbe71401fa733f0d2eb4f45b22df2689f83f5bc 100644
--- a/components/viz/service/frame_sinks/compositor_frame_sink_support.cc
+++ b/components/viz/service/frame_sinks/compositor_frame_sink_support.cc
@@ -1079,6 +1079,11 @@ bool CompositorFrameSinkSupport::ShouldSendBeginFrame(
   // index must be at least as large as our last drawn frame index.
   DCHECK_GE(active_frame_index, last_drawn_frame_index_);
 
+  // Around edit: Our current desktop application renders content using React
+  // portals into child windows. The parent window is hidden which causes some
+  // frames to not be drawn. We ignore this case to continue receiving updates
+  // so that requestAnimationFrame will be called without throttling.
+#if !defined(OS_WIN) && !defined(OS_LINUX)
   // Throttle clients that have submitted too many undrawn frames.
   uint64_t num_undrawn_frames = active_frame_index - last_drawn_frame_index_;
   if (can_throttle_if_unresponsive_or_excessive &&
@@ -1086,6 +1091,7 @@ bool CompositorFrameSinkSupport::ShouldSendBeginFrame(
     RecordShouldSendBeginFrame("ThrottleUndrawnFrames");
     return false;
   }
+#endif
 
   // No other conditions apply so send the begin frame.
   RecordShouldSendBeginFrame("SendDefault");
